frontend/src/components/common/FractionWrapper.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Divider, Input, VStack } from "@chakra-ui/react";
+
+interface FractionWrapperProps {
+  denominator: string;
+  numerator: string;
+  onNumeratorChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onDenominatorChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  readOnly?: boolean;
+}
+
+const getFractionInputWidth = (value: string): number => {
+  return value.length > 1 ? 48 + 8 * value.length : 48;
+};
+
+const getDividerWidth = (width1: number, width2: number) => {
+  return Math.max(width1, width2) + 12;
+};
```

Comment:
Would it be possible to do each of these as a CSS `calc` instead of a Javascript calculation, at least partially? With Chakra UI I've found it's safer to base things off the `rem` unit since that's what they use for everything. That way if we ever had to scale up the UI slightly things would still be the right size. If this is a ton of work though to get right I'm alright leaving it as is

Reply:
found a different way to handle it that's slightly better